driver/compile_common.mli

```diff
@@ -45,28 +45,28 @@ val with_info :
 
 (** {2 Interfaces} *)
 
-val parse_intf : info -> Parsetree.signature
+val parse_intf : info -> Parsetree.interface
 (** [parse_intf info] parses an interface (usually an [.mli] file). *)
 
-val typecheck_intf : info -> Parsetree.signature -> Typedtree.signature
+val typecheck_intf : info -> Parsetree.interface -> Typedtree.interface
 (** [typecheck_intf info parsetree] typechecks an interface and returns
     the typedtree of the associated signature.
 *)
 
-val emit_signature : info -> Parsetree.signature -> Typedtree.signature -> unit
-(** [emit_signature info parsetree typedtree] emits the [.cmi] file
-    containing the given signature.
+val emit_interface : info -> Parsetree.interface -> Typedtree.interface -> unit
+(** [emit_interface info parsetree typedtree] emits the [.cmi] file
+    containing the given interface.
```

Comment:
Compile_common was introduced for compiler-libs users to be able to easily invoke the compiler frontend from their code. I am worried about the backward-compatibility implications of changing the function name here. Could we keep `emit_signature` as an alias of `emit_interface` to increase backward compatibility?

(I'm not sure whether we want to keep the old types for `emit_signature` or use the new types, which is more likely to be passed now that other frontend functions return it. My intuition is that using the new types is fine, but maybe one should look at uses of Compile_common in the wild / on github to tell.)

Reply:
The type change enables adding the location information to cmis, so the type passed to this function will have to change at some point. Unclear whether it should be in this PR or not.